src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -137,20 +137,7 @@ private HttpWindowsProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessi
                             continue;
                         }
 
-                        try
-                        {
-                            // Escape any special characters and unescape * to get wildcard pattern match.
-                            Regex re = new Regex(Regex.Escape(tmp).Replace("\\*", ".*?") + "$",
```

Comment:
It adds "$" at the end but not "^" at the start. Does that mean that it would match suffixes-only, while your new code will not?

Reply:
That's a good catch.  That seems wrong in the previous code.  @wfurt, was that intentional?  The new code could be changed to match by adding a `*` at the beginning of the pattern if necessary.

Reply:
yes, I think this is bug. With curl, .example.com would also match www.example.com (https://curl.haxx.se/libcurl/c/CURLOPT_NOPROXY.html) but for windows it does not work that way.  `*a.com` should match `a.com` and `ba.com` but `a.com` rule should not match `ba.com`

Reply:
Ok, thanks, then I'll leave this as-is. Is fixing it in master sufficient, or should a fix for this (presumably prepending `"^" +`) be considered for 3.1?